test/ruby/CMakeLists.txt

```diff
@@ -13,11 +13,11 @@ macro(add_ruby_test)
         COMMAND "${RUBY_EXECUTABLE}" "-I${CMAKE_BINARY_DIR}/bindings/ruby" "${ARGV1}"
         WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
     )
-    set_property(TEST ${target} PROPERTY ENVIRONMENT
-        "${ASAN_LD_PRELOAD}"
-        "${ASAN_OPTIONS}"
-        "${LSAN_OPTIONS}"
-    )
+
+    # TODO the ODR violation seems like a problem with the Ruby bindings that
+    # should be fixed
+    set(ASAN_OPTIONS "${ASAN_OPTIONS},detect_odr_violation=0")
```

Comment:
Is it setting ASAN_OPTIONS globally or locally?
Maybe a local variable RUBY_ASAN_OPTIONS would be nicer.

Reply:
It's local to the directory scope: https://cmake.org/cmake/help/latest/command/set.html

So should be good.